automation/taskcluster/decision_task_publish.py

```diff
@@ -0,0 +1,224 @@
+            bucket_public_url = 'https://maven-default.stage.mozaws.net/'
+        else:
+            bucket_name = 'maven-production'
+            bucket_public_url = 'https://maven.mozilla.org/'
+
+    scopes = [
+        "project:mobile:android-components:releng:beetmover:bucket:{}".format(bucket_name),
+        "project:mobile:android-components:releng:beetmover:action:push-to-maven",
+    ]
+    return taskcluster.slugId(), BUILDER.beetmover_task(
+        name="Android Components - Publish Module :{} via beetmover".format(artifact_name),
+        description="Publish release module {} to {}".format(artifact_name, bucket_public_url),
+        version=version,
+        artifact_id=artifact_name,
+        dependencies=[build_task_id],
+        upstreamArtifacts=upstream_artifacts,
+        scopes=scopes,
+        is_staging=is_staging,
+        is_snapshot=True
+    )
+
+
+def generate_raw_task(name, description, command_to_run):
+    checkout = ("export TERM=dumb && git fetch {} {} && "
+                "git config advice.detachedHead false && "
+                "git checkout {} && ".format(REPO_URL, BRANCH, HEAD_REV))
+    return taskcluster.slugId(), BUILDER.raw_task(
+        name=name,
+        description=description,
+        command=(checkout +
+                 command_to_run)
+    )
+
+
+def generate_detekt_task():
+    return generate_raw_task(
+        name='Android Components - detekt',
+        description='Running detekt over all modules',
+        command_to_run='./gradlew --no-daemon clean detekt')
+
+
+def generate_ktlint_task():
+    return generate_raw_task(
+        name='Android Components - ktlint',
+        description='Running ktlint over all modules',
+        command_to_run='./gradlew --no-daemon clean ktlint')
+
+
+def generate_compare_locales_task():
+    return generate_raw_task(
+        name='Android Components - compare-locales',
+        description='Validate strings.xml with compare-locales',
+        command_to_run='pip install "compare-locales>=4.0.1,<5.0" && compare-locales --validate l10n.toml .')
+
+
+def generate_and_append_task_to_task_graph(queue, task_graph, generate_function, generate_args=()):
+    task_id, task_definition = generate_function(*generate_args)
+    lib.tasks.schedule_task(queue, task_id, task_definition)
+
+    full_task_definition = queue.task(task_id)
+
+    task_graph[task_id] = {
+        'task': full_task_definition
+    }
+
+    return task_id
+
+
+def release(version, is_snapshot, is_staging):
+    version = lib.module_definitions.get_version_from_gradle() if version is None else version
+
+    queue = taskcluster.Queue({'baseUrl': 'http://taskcluster/queue/v1'})
+
+    artifacts_info = [info for info in lib.module_definitions.from_gradle() if info['shouldPublish']]
+    if len(artifacts_info) == 0:
+        print("Could not get module names from gradle")
+        sys.exit(2)
+
+    task_graph = {}
+
+    for artifact_info in artifacts_info:
```

Comment:
This will parallelizing the builds in releases as well, in the same way they currently are for snapshots. This is going to make @pocmo very happy :)

Reply:
Yes yes yes! 🎉